privacy-dashboard/privacy-dashboard-impl/src/main/java/com/duckduckgo/privacy/dashboard/impl/ToggleReportsDataStore.kt

```diff
@@ -0,0 +1,210 @@
+                jsonAdapter.fromJson(it)
+            }?.dismissInterval ?: DEFAULT_INTERVAL
+        }
+
+    override suspend fun getPromptLimitLogicEnabled(): Boolean =
+        withContext(dispatcherProvider.io()) {
+            return@withContext toggleReportsFeature.self().getSettings()?.let {
+                jsonAdapter.fromJson(it)
+            }?.promptLimitLogicEnabled ?: false
+        }
+
+    override suspend fun getPromptInterval(): Int =
+        withContext(dispatcherProvider.io()) {
+            return@withContext toggleReportsFeature.self().getSettings()?.let {
+                jsonAdapter.fromJson(it)
+            }?.promptInterval ?: DEFAULT_INTERVAL
+        }
+
+    override suspend fun getMaxPromptCount(): Int =
+        withContext(dispatcherProvider.io()) {
+            return@withContext toggleReportsFeature.self().getSettings()?.let {
+                jsonAdapter.fromJson(it)
+            }?.maxPromptCount ?: DEFAULT_MAX_PROMPT
+        }
+
+    override suspend fun insertTogglePromptDismiss() {
+        withContext(dispatcherProvider.io()) {
+            store.edit { prefs ->
+                val currentSet = getPromptsDismissed().toMutableSet()
+                val currentTimeMillis = DatabaseDateFormatter.millisIso8601()
+                val interval = getDismissInterval()
+                currentSet.removeAll { storedTimestamp ->
+                    val storedTime = storedTimestamp.toLong()
+                    (currentTimeMillis - storedTime > interval * 1000L)
+                }
+                val dismissTime = DatabaseDateFormatter.millisIso8601().toString()
+                currentSet.add(dismissTime)
+                prefs[TOGGLE_REPORTS_PROMPTS_DISMISSED] = currentSet
+            }
+        }
+    }
+
+    override suspend fun getPromptsDismissed(): List<String> =
+        withContext(dispatcherProvider.io()) {
+            return@withContext store.data.first().let { prefs ->
+                prefs[TOGGLE_REPORTS_PROMPTS_DISMISSED]?.toList() ?: emptyList()
+            }
+        }
+
+    override suspend fun insertTogglePromptSend() {
+        withContext(dispatcherProvider.io()) {
+            store.edit { prefs ->
+                val currentSet = getReportsSent().toMutableSet()
+                val currentTimeMillis = DatabaseDateFormatter.millisIso8601()
+                val interval = getPromptInterval()
+                currentSet.removeAll { storedTimestamp ->
+                    val storedTime = storedTimestamp.toLong()
+                    (currentTimeMillis - storedTime > interval * 1000L)
+                }
+                val sendTime = DatabaseDateFormatter.millisIso8601().toString()
+                currentSet.add(sendTime)
+                prefs[TOGGLE_REPORTS_SENT] = currentSet
+            }
+        }
+    }
+
+    override suspend fun getReportsSent(): List<String> =
+        withContext(dispatcherProvider.io()) {
+            return@withContext store.data.first().let { prefs ->
```

Comment:
nit: collecting the `Flow` returned by `store.data` doesn't need to be wrapped in `withContext(dispatcherProvider.io()) { }` - DataStore handles switching dispatchers internally.